spec/dfe/analytics/rspec/matchers/have_been_enqueued_event_types_spec.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+require_relative '../../../../../lib/dfe/analytics/rspec/matchers'
+
+class HaveBeenEnqueuedEventTypesTestEvents < ActiveJob::Base
+  def perform(events)
+    Rails.logger.info("performing have_been_enqueued_event_types test event: #{events.inspect}")
+  end
+end
+
+RSpec.describe 'have_been_enqueued_event_types matcher' do
+  before do
+    ActiveJob::Base.queue_adapter = :test
+  end
+
+  let(:web_request_event) { DfE::Analytics::Event.new.with_type('web_request') }
+  let(:update_entity_event) { DfE::Analytics::Event.new.with_type('update_entity') }
+
+  it 'passes when the given event type was triggered' do
+    DfE::Analytics::SendEvents.do([web_request_event.as_json])
+
+    expect(:web_request).to have_been_enqueued_event_types
```

Comment:
This reads a bit oddly — could it be

```ruby
expect_analytics_event_to_have_been_enqueued(:web_request)
```

or even 

```ruby
expect_analytics_web_request_event_to_have_been_enqueued # etc...
```

Reply:
I agree that this reads oddly, I was justifying doing it this way by looking at existing RSpec matchers [`have_been_enqueued`](https://relishapp.com/rspec/rspec-rails/docs/matchers/have-been-enqueued-matcher). It seemed to me that this would provide for the least surprise for users ... but then, maybe you're demonstrating that that's not true? What do you think in light of the existing RSpec matchers?

re: `expect_analytics_web_request_event_to_have_been_enqueued` I did play with having `have_sent_request_analytics_event_types` and `have_sent_entities_analytics_event_types`, but that added complexity at a different levels and I just removed them. Rubocop was whinging about having them in the same file as the `have_sent_analytics_event_types`, and they wouldn't be usable in feature tests such as the [ones I was writing for Publish](https://github.com/DFE-Digital/teacher-training-api/pull/2662/files#diff-34d199d14ec3f3114d2943dc554c19749a671a48ff922d41ac54f61108a35657R47-R49) which would expect both `web_requests` and one of the entity events. 

```ruby
  def then_it_has_sent_analytics_events
    expect([:web_request, :create_entity]).to have_been_enqueued_event_types
  end
```

However, as we've both thought of it, maybe there's something to it. To make this work, I'd think we'd want:

- `expect_analytics_web_request_event_to_have_been_enqueued`
- `expect_analytics_entity_event_to_have_been_enqueued`
- `expect_any_analytics_event_to_have_been_enqueued`

then perhaps for completeness:

- `have_sent_web_request_analytics_events`
- `have_sent_entity_analytics_events`
- `have_sent_any_analytics_events`

and then maybe we could put these into `lib/dfe/analytics/rspec/matchers', or another helper file, to keep Rubocop from whinging.

Or we could just add documentation for the available `event_types` and call it a day. 😉  What do you think?